src/bytes.rs

```diff
@@ -118,14 +119,25 @@ impl<'a> Bytes<'a> {
     pub const fn as_slice(&self) -> &[u8] {
         unsafe { core::slice::from_raw_parts(self.start.as_ptr(), self.len) }
     }
-    #[cfg(feature = "alloc")]
-    /// Proves that the slice can be held onto for arbitrary durations, or copies it into a new one that does.
+    #[cfg(all(feature = "alloc", not(feature = "stabby")))]
+    /// Proves that the slice can be held onto for arbitrary durations, or copies it into a new `Arc<[u8]>` that does.
+    ///
+    /// Notet that `feature = "stabby"` being enabled will cause a `stabby::sync::ArcSlice<u8>` to be used instead of `Arc<[u8]>`.
     pub fn upgrade(self: Bytes<'a>) -> Bytes<'static> {
         if !self.vtable.is_borrowed() {
             return unsafe { core::mem::transmute(self) };
         }
         Arc::<[u8]>::from(self.as_slice()).into()
     }
+
+    #[cfg(feature = "stabby")]
+    /// Proves that the slice can be held onto for arbitrary durations, or copies it into a new [`stabby::sync::ArcSlice<u8>`](stabby::sync::ArcSlice) that does.
+    pub fn upgrade(self: Bytes<'a>) -> Bytes<'static> {
+        if !self.vtable.is_borrowed() {
+            return unsafe { core::mem::transmute(self) };
+        }
+        stabby::sync::ArcSlice::<u8>::from(self.as_slice()).into()
+    }
```

Comment:
To avoid the duplication of docs and signature and `is_borrowed` logic:

```suggestion
        crate::match_cfg! {
            feature = "stabby" => {
                use ::stabby::sync::ArcSlice;
            },
            _ => {
                type ArcSlice<T> = Arc<[T]>;
            },
        }
        ArcSlice::<u8>::from(self.as_slice()).into()
    }
```

EDIT: With a `#[cfg(any(feature = "alloc", feature = "stabby"))]` on top, now (aside: I do wonder whether `no_std` users would have access to C allocators but not `::alloc` 🤔)